WordPress/gutenberg
# Update mechanism that resolves Global Styles data

This PR continues the work started at https://github.com/WordPress/gutenberg/pull/26803 to stabilize the "Global Styles" mechanism in the server, so it can be landed to WordPress core as soon the pieces are ready.

It extracts a new piece that takes care of pulling & merging the data from the different origins that are processed: core, theme, and user. It doesn't change existing behavior.

## How to test

- Make sure existing PHP tests still pass.
- Install and activate a theme with support for theme.json
- Make sure it works as expected in the front-end, site-editor, post-editor.
